### PR TITLE
Init some text about exceptions in services

### PR DESCRIPTION
**Pull request by Ivaylo-Bachvarov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,6 +197,63 @@ class Serializer(serializers.Serializer):
 The implementation of `inline_serializer` can be found in `utils.py` in this repo.
 
 
+## Exception Handling
+
+### Raising Exceptions in Services
+Now we have a great separation between business logic and HTTP logic. The business logic lives in the services and the HTTP logic lives in the APIs.
```
Comment (RadoRado) [inline README.md]: Lets phrase this as:

```
Now we have separation between our HTTP interface & the core logic of our application.
```

File: README.md [hunk 2]
```diff
@@ -197,6 +197,63 @@ class Serializer(serializers.Serializer):
...
+
+In order to keep this separation of concerns our services must not use the `rest_framework.exception` classes because they are bounded with HTTP status codes.
```
Comment (RadoRado) [inline README.md]: Not really sure why are we excluding `rest_framework.exception`, but including `django.core.exceptions` :thinking: 

Should be either - you can use both or use None :thinking: 

Sometimes it's handy to raise `ValidationError` or use the `get_object_or_404` util :thinking: 

Lets discuss

File: README.md [hunk 3]
```diff
@@ -197,6 +197,63 @@ class Serializer(serializers.Serializer):
...
 
+## Exception Handling
+
+### Raising Exceptions in Services
+Now we have a great separation between business logic and HTTP logic. The business logic lives in the services and the HTTP logic lives in the APIs.
+
+In order to keep this separation of concerns our services must not use the `rest_framework.exception` classes because they are bounded with HTTP status codes. 
+Our services must use: `native python exceptions`, `django.core.exceptions` or some custom business exceptions that we define.
+
+Here is a good example of service that preforms some business validation and raise `django.core.exceptions.ValidationError`
```
Comment (RadoRado) [inline README.md]: New line below

File: README.md [hunk 4]
```diff
@@ -197,6 +197,63 @@ class Serializer(serializers.Serializer):
...
+```python
+from django.core.exceptions import ValidationError
+
+def create_topic(*, name: str, course: Course) -> Topic:
+    if Topic.objects.filter(course=course, name=name).exists():
```
Comment (RadoRado) [inline README.md]: This looks like a `unique_together` validation :thinking: Perhaps we need a better example

File: README.md [hunk 5]
```diff
@@ -197,6 +197,63 @@ class Serializer(serializers.Serializer):
...
+        raise ValidationError('Topic with this name already exists for this course!')
+
+    topic = Topic.objects.create(name=name, course=course)
+
+    return topic
+```
+
+### Handle Exceptions in APIs
```
Comment (RadoRado) [inline README.md]: New line below

File: README.md [hunk 6]
```diff
@@ -197,6 +197,63 @@ class Serializer(serializers.Serializer):
...
+    def handle_exception(self, exc):
+        if isinstance(exc, tuple(self.expected_exceptions.keys())):
+            drf_exception_class = self.expected_exceptions[exc.__class__]
+            drf_exception = drf_exception_class()
+
+            return super().handle_exception(drf_exception)
+
+        return super().handle_exception(exc)
+```
+You can move this code to a mixin and use it in every API to prevent code duplication.
```
Comment (RadoRado) [inline README.md]: Lets add an example mixin for that & also add a code sample in `utils.py`

File: README.md (hunk 2 above)
Comment (Ivaylo-Bachvarov) [reply README.md]: Up for discussion on Monday.

**Pull request merged**